### PR TITLE
Bugfix/nept 106

### PR DESCRIPTION
**Pull request by gillesdeudon**

### CODE REVIEW

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module [hunk 1]
```diff
@@ -264,10 +264,8 @@ function _nexteuropa_multilingual_alias_node_form_alter(&$form, &$form_state) {
       // 1. If pathauto deactivated, defining the right default value.
```
Comment (cyberwolf) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Is this comment still valid? We don't care about pathauto being deactivated any longer in the code below.

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module [hunk 2]
```diff
@@ -298,26 +296,27 @@ function _nexteuropa_multilingual_alias_node_form_alter(&$form, &$form_state) {
  *   - alias: The the LANGUAGE_NONE alias.
  *   - pathauto: The flag indicating if pathauto is activated for the node.
  */
-function _nexteuropa_multilingual_get_und_entity_alias_info($entity, $entity_type) {
+function _nexteuropa_multilingual_get_und_entity_alias_info($entity_type, $entity) {
```
Comment (cyberwolf) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Is there a particular reason why you switched the order of the arguments? I would leave it like it was, thus providing a patch for the bug and nothing more, reducing the diff and limiting risks.

File: tests/features/content_editing_editor.feature [hunk 3]
```diff
@@ -37,4 +37,21 @@ Feature: Content editing as editor
       | <blockquote>Lorem ipsum dolor sit amet.</blockquote>                                                | <blockquote><p>Lorem ipsum dolor sit amet.</p></blockquote>                                         |
       | <p>Lorem ipsum <code>dolor(sit);</code> amet.</p>                                                   | <p>Lorem ipsum <code>dolor(sit);</code> amet.</p>                                                   |
       | <ul><li>Lorem ipsum</li><li>Dolor sit amet</li></ul>                                                | <ul><li>Lorem ipsum</li>                                                                            |
-      | <ol><li>Lorem ipsum</li><li>Dolor sit amet</li></ol>                                                | <ol><li>Lorem ipsum</li>                                                                            |
+      | <ol><li>Lorem ipsum</li><li>Dolor sit amet</li></ol>                                                | <ol><li>Lorem ipsum</li>
+
+  @api
+  Scenario: User can create an article but he cannot define it path alias, even during an update.
```
Comment (cyberwolf) [inline tests/features/content_editing_editor.feature]: **its** path alias

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module (hunk 2 above)
Comment (gillesdeudon) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Yes, I realized that this code I've done in a previous ticket was not aligned on the majority of API existing in Drupal and Entity module. 
Then, I aligned the function signature on what I observed to ease the code reading. As it is a private function and it is only used once so far, risks related to execution time errors are in existent.

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module (hunk 1 above)
Comment (gillesdeudon) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Right! :smile:

**Comment by cyberwolf** [comment]
👍

**Comment by Fefaine** [comment]
Needs more work for cases where when various patterns are available.
My suggestion would be to disable (& empty) the pattern for all languages that are not UND for entity translation content + add a description message

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module [hunk 4]
```diff
@@ -298,26 +300,30 @@ function _nexteuropa_multilingual_alias_node_form_alter(&$form, &$form_state) {
...
   list(, , $bundle) = entity_extract_ids($entity_type, $entity);
   $pattern = pathauto_pattern_load_by_entity($entity_type, $bundle, LANGUAGE_NONE);
   if (empty($pattern)) {
     return FALSE;
   }
   // Get the corresponding LANGUAGE_NONE alias.
-  $entity_uri = $entity->path['source'];
-  $language_none_path = path_load(array('source' => $entity_uri, 'language' => LANGUAGE_NONE));
+  $entity_uri = entity_uri($entity_type, $entity);
+  $entity_path = $entity_uri['path'];
```
Comment (verbruggenalex) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Please account for the possibility of entity_uri() returning the value NULL.

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module [hunk 5]
```diff
@@ -298,26 +300,30 @@ function _nexteuropa_multilingual_alias_node_form_alter(&$form, &$form_state) {
...
+  $language_none_path = path_load(array('source' => $entity_path, 'language' => LANGUAGE_NONE));
   $language_none_alias = $language_none_path['alias'];
```
Comment (verbruggenalex) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Please account for the possibility of path_load() returning the value FALSE.

File: profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module [hunk 6]
```diff
@@ -261,12 +261,14 @@ function _nexteuropa_multilingual_alias_node_form_alter(&$form, &$form_state) {
...
     if ($set_translation_mechanism == ENTITY_TRANSLATION_ENABLED) {
-      // 1. If pathauto deactivated, defining the right default value.
-      // reproduction of the default behavior but with the LANGUAGE_NONE alias.
+      // 1.  Reproduction of the default behavior but with the LANGUAGE_NONE
+      // alias.
       if ($nid) {
-        $und_alias_info = _nexteuropa_multilingual_get_und_entity_alias_info($node, 'node');
-        if (!$und_alias_info['pathauto']) {
-          $form['path']['alias']['#default_value'] = $und_alias_info['alias'];
+        $und_alias_info = _nexteuropa_multilingual_get_und_entity_alias_info('node', $node);
```
Comment (verbruggenalex) [inline profiles/common/modules/features/nexteuropa_multilingual/nexteuropa_multilingual.module]: Please account for the possibility of the function returning FALSE.

**Comment by verbruggenalex** [comment]
:+1:

**Comment by Fefaine** [comment]
:+1:

**Comment by gillesdeudon** [comment]
@Fefaine ,

That's why I put it in wip because it must be reviewed or replace in order to not block tickets that do not have any links with security concerns.

It is not the purpose of a business test to perform an operational check.

File: tests/features/change_tracking_workflow.feature [hunk 7]
```diff
@@ -111,12 +127,19 @@ Feature: Change tracking features
...
+    And the following languages are available:
       | languages |
       | en        |
       | fr        |
+    When I go to "admin/config/content/multisite_wysiwyg/workbench"
+    And I check the box "edit-nexteuropa-editorial-tracking-wbm-states-validated"
+    And I check the box "edit-nexteuropa-editorial-tracking-wbm-states-published"
+    And I press "Save configuration"
+    Then I should see the success message "The configuration options have been saved."
     When I am viewing a multilingual "page" content:
```
Comment (imanoleguskiza) [inline tests/features/change_tracking_workflow.feature]: I'm not sure I understand why this changes are made in this pull request. They don't seem to belong to this ticket at all?

**Comment by imanoleguskiza** [comment]
There are changes to multiple behat tests that appear to have little relationship with the ticket we are trying to solve. Is this pull request the right place to do this changes? (I'm referring to the changes in change_tracking_workflow.feature, embedded_image.feature and frontend_smoke_test.feature)

**Comment by gillesdeudon** [comment]
@imanoleguskiza 
Concerning your remarks related to Behat tests, these changes were required because of merge of the master into the branch. I had no choice but doing these changes. 
Normally, everything will be normalized with the final merge into the master.

File: tests/features/change_tracking_workflow.feature (hunk 7 above)
Comment (gillesdeudon) [inline tests/features/change_tracking_workflow.feature]: They comes from the merge of the master into the branch.

**Comment by imanoleguskiza** [comment]
👍

**Comment by netlooker** [comment]
👍
